packages/core/src/remoteconfig.ts

```diff
@@ -0,0 +1,146 @@
+import type {
+  RemoteOverrideableConfig,
+  RemoteConfigOptions,
+  RemoteConfigPayload,
+  RemoteConfigInterface,
+  RemoteConfigSource,
+} from '@sentry/types';
+
+const DEFAULT_CONFIG_NAME = '__default';
+
+/**
+ * Remote Configuration fetches configuration from a remote server.
+ */
+export function remoteConfig({
+  defaultConfigName = DEFAULT_CONFIG_NAME,
+  storage,
+  transport,
+}: RemoteConfigOptions): RemoteConfigInterface {
+  let _activeConfig: RemoteConfigPayload | undefined;
+  let _pendingConfig: RemoteConfigPayload | undefined;
+  let _lastFetch: Date | undefined;
+  let _lastUpdate: Date | undefined;
+  let _source: RemoteConfigSource = 'DEFAULT';
+  let _hasUpdate: boolean = false;
+  let _state: RemoteConfigState = 'INITIALIZING';
+
+  console.log('remoteConfig!!');
+
+  function _initConfig() {
+    // Use cached configuration if it exists
+    const cachedValue = storage.get(defaultConfigName);
+    if (cachedValue) {
+      _activeConfig = JSON.parse(cachedValue);
```

Comment:
I think we should be putting JSON.parse/serizlize into the storage object itself, and it's going to need try/catch guards around it